examples/protocol-and-stream-muxing/README.md

```diff
@@ -6,155 +6,161 @@ The feature of agreeing on a protocol over an established connection is what we
 
 # 1. Handle multiple protocols
 
-Let's see _protocol multiplexing_ in action! You will need the following modules for this example: `libp2p`, `libp2p-tcp`, `peer-info`, `async` and `pull-stream`. This example reuses the base left by the [Transports](../transports) example. You can see the complete solution at [1.js](./1.js).
+Let's see _protocol multiplexing_ in action! You will need the following modules for this example: `libp2p`, `libp2p-tcp`, `peer-info`, `it-pipe`, `it-buffer` and `streaming-iterables`. This example reuses the base left by the [Transports](../transports) example. You can see the complete solution at [1.js](./1.js).
 
 After creating the nodes, we need to tell libp2p which protocols to handle.
 
 ```JavaScript
+const pipe = require('it-pipe')
+const { map } = require('streaming-iterables')
+const { toBuffer } = require('it-buffer')
+
 // ...
 const node1 = nodes[0]
 const node2 = nodes[1]
 
-// Here we are telling libp2p that is someone dials this node to talk with the `/your-protocol`
-// multicodec, the protocol identifier, please call this callback and give it the connection
+// Here we are telling libp2p that if someone dials this node to talk with the `/your-protocol`
+// multicodec, the protocol identifier, please call this handler and give it the stream
 // so that incomming data can be handled
-node2.handle('/your-protocol', (protocol, conn) => {
-  pull(
-    conn,
-    pull.map((v) => v.toString()),
-    pull.log()
+node2.handle('/your-protocol', ({ stream }) => {
+  pipe(
+    stream,
+    toBuffer,
+    map(String),
+    source => (async function () {
+      for await (const msg of source) {
+        console.log(msg)
+      }
+    })()
   )
 })
```

Comment:
This can be:
```js
node2.handle('/your-protocol', ({ stream }) => {
  pipe(
    stream,
    async function (source) {
      for await (const msg of source) {
        console.log(msg.toString())
      }
    }
  )
})
```